vm/devices/storage/storage_tests/tests/scsidvd_nvme.rs

```diff
@@ -0,0 +1,226 @@
+            .add_namespace(
+                1,
+                disklayer_ram::ram_disk(sector_size as u64 * sector_count, read_only).unwrap(),
+            )
+            .await
+            .unwrap();
+
+        let device = EmulatedDevice::new(nvme, msi_set, mem);
+        let nvme_driver = NvmeDriver::new(&driver_source, CPU_COUNT, device)
+            .await
+            .unwrap();
+        let namespace = nvme_driver.namespace(1).await.unwrap();
+        let buf_range = OwnedRequestBuffers::linear(0, 16384, true);
+        for i in 0..(sector_count / 8) {
+            namespace
+                .write(
+                    0,
+                    i * 8,
+                    8,
+                    false,
+                    &payload_mem,
+                    buf_range.buffer(&payload_mem).range(),
+                )
+                .await
+                .unwrap();
+        }
+        let disk = NvmeDisk::new(namespace);
+
+        let scsi_dvd = SimpleScsiDvd::new(Some(Disk::new(disk).unwrap()));
+        Self {
+            scsi_dvd,
+            _nvme_driver: nvme_driver,
+        }
+    }
+}
+
+fn make_repeat_data_buffer(sector_count: usize, sector_size: usize) -> Vec<u8> {
+    let mut buf = vec![0u8; sector_count * sector_size];
+    let mut temp = vec![0u8; sector_size];
+    assert!(sector_size > 2);
+    temp[sector_size / 2 - 1] = 2;
+    temp[sector_size / 2] = 3;
+
+    for i in (0..buf.len()).step_by(temp.len()) {
+        let end_point = i + temp.len();
+        buf[i..end_point].copy_from_slice(&temp);
+    }
+
+    buf
+}
+
+async fn check_execute_scsi(
+    scsi_dvd: &mut SimpleScsiDvd,
+    external_data: &RequestBuffers<'_>,
+    request: &Request,
+    pass: bool,
+) {
+    let result = scsi_dvd.execute_scsi(external_data, request).await;
+    match pass {
+        true if result.scsi_status != scsi_defs::ScsiStatus::GOOD => {
+            panic!(
+                "execute_scsi failed! request: {:?} result: {:?}",
+                request, result
+            );
+        }
+        false if result.scsi_status == scsi_defs::ScsiStatus::GOOD => {
+            panic!(
+                "execute_scsi passed! request: {:?} result: {:?}",
+                request, result
+            );
+        }
+        _ => (),
+    }
+}
+
+fn make_cdb16_request(operation_code: ScsiOp, start_lba: u64, lba_count: u32) -> Request {
+    let cdb = scsi_defs::Cdb16 {
+        operation_code,
+        flags: scsi_defs::Cdb16Flags::new(),
+        logical_block: start_lba.into(),
+        transfer_blocks: lba_count.into(),
+        reserved2: 0,
+        control: 0,
+    };
+    let mut data = [0u8; 16];
+    data[..].copy_from_slice(cdb.as_bytes());
+    Request {
+        cdb: data,
+        srb_flags: 0,
+    }
+}
+
+fn check_guest_memory(
+    guest_mem: &GuestMemory,
+    start_lba: u64,
+    buff: &[u8],
+    sector_size: usize,
+) -> bool {
+    let mut b = vec![0u8; buff.len()];
+    if guest_mem.read_at(start_lba, &mut b).is_err() {
```

Comment:
The output will be more helpful if you simplify this down to:

`guest_mem.read_at(start_lba, &mut b).expect("guest_mem read error")`

You'll get the backtrace of the `Err(..)` Result.